File: 786/cpp/main.cpp
#include<iostream>
using namespace std;

const int N = 1e5+10;
int n = 0;
int k = 0;
int a[N];

void QuickSort(int a[], int l, int r) {
    if ( l >= r ) {
        return;
    }
    int key = a[(l+r) >> 1];
    int i = l-1;
    int j = r+1;
    while ( i < j ) {
        do i++; while(a[i] < key);
        do j--; while(a[j] > key);
        if (i<j) {
            swap(a[i], a[j]);
        }
    }
    QuickSort(a, l, j);
    QuickSort(a, j+1, r);
}

int main() {
    scanf("%d",&n);
    scanf("%d", &k);
    for ( int i = 0; i < n ; i++ ) {
        scanf("%d", &a[i]);
    }
    QuickSort(a, 0 , n - 1);
    printf("%d\n", a[k-1]);
}

File: 786/go/main.go
package main

import "fmt"

func QuickSort(a []int, l, r int) {
	if l >= r {
		return
	}
	key, i, j := a[(l+r)>>1], l-1, r+1
	for i < j {
		i++
		for a[i] < key {
			i++
		}
		j--
		for a[j] > key {
			j--
		}
		if i < j {
			a[i], a[j] = a[j], a[i]
		}
	}
	QuickSort(a, l, j)
	QuickSort(a, j+1, r)
}

func main() {
	n, k := 0, 0
	fmt.Scanf("%d", &n)
	fmt.Scanf("%d", &k)
	a := make([]int, 0)
	for i := 0; i < n; i++ {
		x := 0
		fmt.Scan(&x)
		a = append(a, x)
	}
	QuickSort(a, 0, n-1)
	fmt.Println(a[k-1])
}
